services/src/main/java/org/fao/geonet/api/userfeedback/UserFeedbackAPI.java

```diff
@@ -0,0 +1,424 @@
+            response.setStatus(HttpStatus.FORBIDDEN.value());
+            return null;
+        }
+
+        Log.debug("org.fao.geonet.api.userfeedback.UserFeedback", "getUserComment");
+
+        final IUserFeedbackService userFeedbackService = (IUserFeedbackService) ApplicationContextHolder.get()
+                .getBean("userFeedbackService");
+
+        final UserSession session = ApiUtils.getUserSession(httpSession);
+
+        boolean published = true; // Takes only published comments
+
+        // showing not published comments only to logged users (maybe better
+        // restrict to Reviewers)
+        if (session != null && session.isAuthenticated()) {
+            published = false;
+        }
+
+        final UserFeedback userfeedback = userFeedbackService.retrieveUserFeedback(uuid, published);
+
+        UserFeedbackDTO dto = null;
+
+        if (userfeedback != null) {
+            dto = UserFeedbackUtils.convertToDto(userfeedback);
+        }
+
+        // Check permission for metadata
+        final Metadata metadata = ApiUtils.canViewRecord(userfeedback.getMetadata().getUuid(), request);
+        if (metadata == null) {
+            printOutputMessage(response, HttpStatus.FORBIDDEN, ApiParams.API_RESPONSE_NOT_ALLOWED_CAN_VIEW);
+            return null;
+        }
+
+        return dto;
+    }
+
+    /**
+     * Gets the user comments.
+     *
+     * @param request the request
+     * @param response the response
+     * @param httpSession the http session
+     * @return the user comments
+     * @throws Exception the exception
+     */
+    // GET
+    @ApiOperation(value = "Finds a list of user feedback records. ", 
+            notes = " This list will include also the draft uf if the client is logged as reviewer. "
+            + " target={metadata uuid} maxnumber={max result size} "
+            , nickname = "getUserComments")
+    @RequestMapping(value = "/userfeedback", produces = MediaType.APPLICATION_JSON_VALUE, method = RequestMethod.GET)
+    @ResponseStatus(value = HttpStatus.OK)
+    @ResponseBody
+    public List<UserFeedbackDTO> getUserComments(final HttpServletRequest request, final HttpServletResponse response,
+            final HttpSession httpSession) throws Exception {
+
+        final ApplicationContext appContext = ApplicationContextHolder.get();
+        final SettingManager settingManager = appContext.getBean(SettingManager.class);
+        final String functionEnabled = settingManager.getValue(Settings.SYSTEM_LOCALRATING_ENABLE);
+
+        if (!functionEnabled.equals(RatingsSetting.ADVANCED)) {
+            response.setStatus(HttpStatus.FORBIDDEN.value());
+            return null;
+        }
+
+        try {
+            Log.debug("org.fao.geonet.api.userfeedback.UserFeedback", "getUserComments");
+
+            final IUserFeedbackService userFeedbackService = getUserFeedbackService();
+
+            final String uuid = request.getParameter("target");
```

Comment:
```target``` and ```maxnumber``` parameters are not part of the method signature and they are not displayed in the service doc 
![image](https://user-images.githubusercontent.com/1701393/38668930-5b795892-3e45-11e8-9186-c3941c1a8da8.png)

Use ```@RequestParam``` annotation ?

Also "target" may be renamed to "record" maybe as it is a metadata record.

At some point we may need to add paging here - but this is not part of this PR - we should maybe plan for adding this for users/groups/... API.

Reply:
If target is mandatory, should we move that service to /records/{uuid}/userfeedback ? would be more consistent with userfeedbackrating

Reply:
target is not mandatory, so maybe not. or should we have both ?

Reply:
None of them is mandatory. I added them (uuid, previously target, and maxnumber) as not requested parameters